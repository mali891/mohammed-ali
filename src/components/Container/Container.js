import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { styles } from './Container.styles'

const Container = ({ children, className, ...other }) => {
  const dynamicClassNames = classNames([className], {
    'c-container': true
  })

  return (
    <div css={styles} className={dynamicClassNames} {...other}>
      {children}
    </div>
  )
}

Container.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
}

export default Container
