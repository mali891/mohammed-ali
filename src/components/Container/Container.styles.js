import { css } from '@emotion/core'

export const styles = css`
	margin: 0 auto;
	width: auto;
	max-width: 100rem;
`
