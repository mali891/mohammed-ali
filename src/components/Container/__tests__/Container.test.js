import React from 'react'
import { render } from '@testing-library/react'

import Container from '../Container'

describe('Container', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Container />)
    expect(asFragment()).toMatchSnapshot()
  })
})
