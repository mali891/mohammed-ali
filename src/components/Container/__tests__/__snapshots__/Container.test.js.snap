// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<DocumentFragment>
  <div
    class="c-container"
    css="[object Object]"
  />
</DocumentFragment>
`;
